libglusterfs/src/common-utils.c

```diff
@@ -4283,3 +4280,21 @@ gf_gethostname(void)
 {
     return global_ctx->hostname;
 }
+
+rlim_t
+gf_set_nofile(rlim_t high, rlim_t low)
+{
+    int n;
+    struct rlimit lim;
+    rlim_t r[2] = { high, low };
+
+    for (n = 0; n < 2; n++)
+        if (r[n] != 0) {
+            lim.rlim_cur = r[n];
+            lim.rlim_max = r[n];
+            if (setrlimit(RLIMIT_NOFILE, &lim) == 0)
+                return r[n];
+        }
+
+    return RLIM_INFINITY;
```

Comment:
This could potentially be a valid value (i.e. there's no predefined limit). Wouldn't it be better to just return a boolean or 0/-1 instead of an `rlim_t` ?

Reply:
Voting for void return value, see below.